src/util/Ip.ts

```diff
@@ -60,4 +60,18 @@ export function onSameNetwork(ip1: string, ip2: string, mask: string) {
             if ((ip1Array[i] & mask) !== (ip2Array[i] & mask)) return false;
         }
     }
+    return false
 }
+
+
+/**
+ * return true if ipV6 address meets the Matter Spec to advertise in MDNS
+ * filter out IPv6 Loopback addresses - fe80 -  for obvious reasons
+ */
+export function isMatterAddressableIPv6Address(targetIpAddr: string): boolean {
+    return !(
+        targetIpAddr.toLowerCase().startsWith("fe80") ||
```

Comment:
@vves Hm ... why you now sort out link local too? fe80 are all link locals ... why not announce them? In my eyes they are valid

Reply:
I have come to the conclusion based on empirical testing and reading the spec that while fe80 is valid, it is NOT valid when it's the ONLY address on an interface.  The larger issue that I have come across is that if an interface only provides on they IPV6 fe80 address the HomeApp and SmartThings will not resolve the endpoint.  The solution is to refactor to validate the interface provides at least ONE valid IPv6 address that is not a loopback address or IPv4 Address, but NOT remove the loopback and IPv4 Addresses.  If there is not at least one Matter addressable IPv6 address, the library should not advertise that interface.  This is the list from my test machine where only the en0 interface is addressable from HomeApp or SmartThings.  This function is valid for as named, but the use of it needs to be in conjuntion with a network interface validator.
```2023-03-15 09:39:46.335 DEBUG MdnsBroadcaster anpi1 on IP Addrs: fe80::5082:3eff:fe47:1cdd host: 52823E471CDD0000.local
2023-03-15 09:39:46.336 DEBUG MdnsBroadcaster anpi0 on IP Addrs: fe80::5082:3eff:fe47:1cdc host: 52823E471CDC0000.local
2023-03-15 09:39:46.336 DEBUG MdnsBroadcaster anpi2 on IP Addrs: fe80::5082:3eff:fe47:1cde host: 52823E471CDE0000.local
2023-03-15 09:39:46.337 DEBUG MdnsBroadcaster en0 on IP Addrs: fe80::85e:cc32:9f1:c265, fd69:48a4:185f:1:10eb:e991:a821:617e, 2601:285:500:bf9:14a5:cb4d:2f38:873f, 2601:285:500:bf9:3885:23da:3b72:902d, fd68:9b7f:6836:44cd:25:de98:15d:375c, 10.0.10.145 host: F02F4B09EDFA0000.local
2023-03-15 09:39:46.337 DEBUG MdnsBroadcaster awdl0 on IP Addrs: fe80::80ab:7dff:fe1a:4df host: 82AB7D1A04DF0000.local
2023-03-15 09:39:46.338 DEBUG MdnsBroadcaster llw0 on IP Addrs: fe80::80ab:7dff:fe1a:4df host: 82AB7D1A04DF0000.local
2023-03-15 09:39:46.338 DEBUG MdnsBroadcaster utun0 on IP Addrs: fe80::9ae3:6419:5975:cc9 host: 0000000000000000.local
2023-03-15 09:39:46.339 DEBUG MdnsBroadcaster utun1 on IP Addrs: fe80::97e2:cfd3:445a:8763 host: 0000000000000000.local
2023-03-15 09:39:46.339 DEBUG MdnsBroadcaster utun2 on IP Addrs: fe80::ce81:b1c:bd2c:69e host: 0000000000000000.local
2023-03-15 09:39:46.340 DEBUG MdnsBroadcaster utun3 on IP Addrs: fe80::e81a:e66d:611a:b465 host: 0000000000000000.local
2023-03-15 09:39:46.340 DEBUG MdnsBroadcaster utun4 on IP Addrs: fe80::5120:d18f:4b0e:ed81 host: 0000000000000000.local
2023-03-15 09:39:46.341 DEBUG MdnsBroadcaster utun5 on IP Addrs: fe80::1c2a:4bb3:bf2:39b4 host: 0000000000000000.local
2023-03-15 09:39:46.341 DEBUG MdnsBroadcaster utun6 on IP Addrs: fe80::557c:6550:d754:c9c host: 0000000000000000.local
2023-03-15 09:39:46.341 DEBUG MdnsBroadcaster utun7 on IP Addrs: fe80::f4d0:e8d5:2bde:f628 host: 0000000000000000.local
2023-03-15 09:39:46.342 DEBUG MdnsBroadcaster utun8 on IP Addrs: fe80::b396:3357:d6aa:fd8e host: 0000000000000000.local```